### PR TITLE
[ENG-1188] prepare network large countries

### PR DESCRIPTION
**Pull request by ollie-bell**
* incorporated from upstream #845 optional kdtree for build_osm_network - enables reasonable runtime for US.
* added functionality to clip country geometry at high latitudes and at the antimeridian.
* completed prepare network for CA, US, NZ
* other reorganisation

### CODE REVIEW

**Comment by ollie-bell** [comment]
Also closes tickets ENG-1258 and ENG-1259

File: _TRACE_outputs/progress-map.png
Comment (CalvinNesbitt) [inline _TRACE_outputs/progress-map.png:1]: Nice 😎

File: country_configs/config.US.yaml [hunk 1]
```diff
@@ -104,6 +104,7 @@ build_shape_options:
   worldpop_method: "standard"  # "standard" pulls from web 1kmx1km raster, "api" pulls from API 100mx100m raster, false (not "false") no pop addition to shape which is useful when generating only cutout
   gdp_method: "standard"  # "standard" pulls from web 1x1km raster, false (not "false") no gdp addition to shape which useful when generating only cutout
   contended_flag: "set_by_country" # "set_by_country" assigns the contended areas to the countries according to the GADM database, "drop" drops these contended areas from the model
+  clip_antimeridian: "west"
```
Comment (CalvinNesbitt) [inline country_configs/config.US.yaml:107]: 👏

File: country_configs/config.CA.yaml [hunk 2]
```diff
@@ -197,7 +198,7 @@ transformers:
   type: ""
 
 atlite:
-  nprocesses: 1
+  nprocesses: 8
```
Comment (CalvinNesbitt) [inline country_configs/config.CA.yaml:201]: This is just for speed?/making better use of dask

File: scripts/build_cutout.py [hunk 3]
```diff
@@ -126,7 +127,9 @@
         offshore = gpd.read_file(offshore_shapes)
         regions = pd.concat([onshore, offshore])
         d = max(cutout_params.get("dx", 0.25), cutout_params.get("dy", 0.25)) * 2
-        cutout_params["bounds"] = regions.total_bounds + [-d, -d, d, d]
+        cutout_params["bounds"] = np.clip(
```
Comment (CalvinNesbitt) [inline scripts/build_cutout.py:130]: minor: add flag based on snakemake arg

**Review by CalvinNesbitt — APPROVED** [review]

**Pull request merged**